### DISCUSSION TITLE
Is there a way to reload Brython within the page ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2187] [Q&A] [answered]
Hi everyone,

Brython load trough 
```html
<body onload="brython()">
```
Is there a **anyway** to trigger a reload ? even with JS 🤮 ?

**Comment by BobSquarePants** [comment]
from Brython
```python
window.brython()
```
works, but when call once more the SCRIPT call with
```html
<script type="text/python" src="/aBrythonFile.bry"></script> 
```
are loaded twice, and if `window.brython()` is called again then thair are load 4x and so on..

**Comment by PierreQuentel** [comment, marked as answer]
Hi,

You can specify the scripts that you want to "reload" by specifying a key "ids" in the call to `brython()`:

```xml
<script type="text/python">
from browser import bind, window

print('main script')

@bind('#test', 'click')
def test(ev):
    window.brython({'ids': ['reload']}) # only run the script with id "reload"

</script>

<script type="text/python" id="reload">
print('reload')
</script>

<button id="test">reload script</button>
```

**Reply by BobSquarePants** [reply]
Thank you @PierreQuentel ,

**PART ONE: SOLVED**

This is neat indeed !, however, your example work with an `click` event from an already loaded Brython.

If my page receive new content from (for example) an ajax return. 
within that returned content is possible to initiate a reload too ?

currently I'm using 
```html
<script type="text/javascript">brython(1)</script>
```
in the returned content. it work but it seem not to be optimal (trigger twice some events ?)

---
**PART TWO**

out of topic, from your example I see that Brython now have decorators 🏆 
```python
@bind('#test', 'click')
def test(ev):
    ......
```
This is great ! Brython is the future !!
are decorators already in place in [v 3.11.2](https://github.com/brython-dev/brython/releases/tag/3.11.2) ?

**Reply by BobSquarePants** [reply]
Nevermind the first part ! I've found a way ! Thx to your example !

**Comment by PierreQuentel** [comment]
> out of topic, from your example I see that Brython now have decorators 🏆

Decorators have been supported almost from the beginning of Brython ;-) The `bind` decorator was added in Brython 3.6.0, published in June 2018.

**Reply by BobSquarePants** [reply]
Hoo great news, I didn't know !

Can we use it in combination of a `.select()` ?

```python
@bind(document.select('#anIDedel+ div')[0], 'click')
def test(ev):
    ......
```

?

**Reply by BobSquarePants** [reply]
FYI, I just tried to use `@bind()`

but this give me 
```js
@bind('#something', 'change')
^^^^
NameError: name 'bind' is not defined
```
Brython v3.11.0

**Reply by PierreQuentel** [reply]
`bind()` is in module **`browser`** (see the [documentation](https://www.brython.info/static_doc/fr/events.html))

**Reply by BobSquarePants** [reply]
indeed. I will take the time to re-read all of it. Thanks again !
